storage/pebble/register_pruner_run.go

```diff
@@ -0,0 +1,120 @@
+package pebble
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// DBPruner is responsible for pruning outdated register entries in a Pebble database.
+// It batches deletions to improve performance and reduce the load on the system.
+type DBPruner struct {
+	logger zerolog.Logger
+	// pruneThrottleDelay controls a small pause between batches of registers inspected and pruned
+	pruneThrottleDelay time.Duration
+	pruneHeight        uint64
+
+	db                   *pebble.DB
+	lastRegisterID       flow.RegisterID
+	keepFirstRelevantKey bool
+
+	totalKeysPruned int
+}
+
+func RegisterPrunerRun(db *pebble.DB, logger zerolog.Logger, pruneThrottleDelay time.Duration, pruneHeight uint64) *DBPruner {
```

Comment:
my comment about naming wasn't specific to the function, I was talking about the struct. I'm not attached to the name. I just want the functionality of this to be clearly indicated in the name. 

currently we have `RegisterPruner` and `DBPruner`. Based on the names, I would assume one pruned the register db, and the other was a general purpose pruner, but they had basically the same functionality. As implemented, `RegisterPruner` is the high level pruner, and `DBPruner` handles operations for an individual pruning cycle.